### ISSUE TITLE
Logging handler namer function has the wrong annotations

### ISSUE THREAD

**Issue by vim345** [python/typeshed#1497]
I have the following code:

```
import logging

def _namer(name: str) -> str:
    return name + '.gz'

def get_logger(file_name: str, level: str) -> logging.Logger:
    logger = logging.getLogger(__name__)
    handler = TimedRotatingFileHandler(file_name, backupCount=30, when='midnight')

    handler.namer = _namer
    return logger
```

When I run mypy on it, I get this error:
error: Incompatible types in assignment (expression has type Callable[[str], str], variable has type "Optional[Callable[[str], None]]")

I believe the type for namer should be changed to

```Callable[[str], str]```

over here:

https://github.com/python/typeshed/blob/master/stdlib/2and3/logging/handlers.pyi#L32

**Comment by JelleZijlstra** [comment]
Thanks for the report! Could you submit a PR?

**Comment by ceh** [comment]
I think issue was fixed by #1498; we can close this issue. (cc @JelleZijlstra)

**Comment by JelleZijlstra** [comment]
yes, thanks
